backend/src/main/java/org/eclipse/tractusx/puris/backend/controller/MaterialPartnerRelationsController.java

```diff
@@ -0,0 +1,148 @@
+import org.eclipse.tractusx.puris.backend.masterdata.domain.model.Partner;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialPartnerRelationService;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.MaterialService;
+import org.eclipse.tractusx.puris.backend.masterdata.logic.service.PartnerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("materialpartnerrelations")
+public class MaterialPartnerRelationsController {
+
+
+    @Autowired
+    private MaterialService materialService;
+
+    @Autowired
+    private PartnerService partnerService;
+
+    @Autowired
+    private MaterialPartnerRelationService mprService;
+
+    @PostMapping
+    @CrossOrigin
+    @Operation(description = "Creates a new MaterialPartnerRelation with the given parameter data. " +
+        "Please note that this is only possible, if the designated Material " +
+        "and Partner entities have already been created before this request. ")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Successfully created a new MaterialPartnerRelationEntity."),
+        @ApiResponse(responseCode = "400", description = "Material and/or Partner do not exist."),
+        @ApiResponse(responseCode = "409", description = "Relation for given Material and Partner does already exist."),
+        @ApiResponse(responseCode = "500", description = "Internal Server Error")
+    })
+    public ResponseEntity<?> createMaterialPartnerRelation(
+        @Parameter(name = "ownMaterialNumber", description = "The Material Number that is used in your own company to identify the Material.",
+            example = "MNR-7307-AU340474.002", required = true) @RequestParam String ownMaterialNumber,
+        @Parameter(name = "partnerBpnl", description = "The unique BPNL that was assigned to that Partner.",
+            example = "BPNL2222222222RR", required = true) @RequestParam() String partnerBpnl,
+        @Parameter(name = "partnerMaterialNumber", description = "The Material Number that this Partner is using in his own company to identify the Material.",
+            example = "MNR-8101-ID146955.001") @RequestParam String partnerMaterialNumber,
+        @Parameter(name = "partnerSupplies", description = "This boolean flag indicates whether this Partner is a potential supplier of the given Material.",
+            example = "true") @RequestParam boolean partnerSupplies,
+        @Parameter(name = "partnerBuys", description = "This boolean flag indicates whether this Partner is a potential customer of this Material.",
+            example = "true") @RequestParam boolean partnerBuys) {
+        Material material = materialService.findByOwnMaterialNumber(ownMaterialNumber);
```

Comment:
Does the repository interface can handle null values for ownMaterialNumber, that is mandatory but not checked for null values?

Reply:
The Spring framework makes sure that ownMaterialNumber cannot be null here, since it automatically fires an exception and returns 400 if a request comes in that is missing one or more required request parameters. 
Additionally I realized now that the "required"-attribute defaults to true, so I'd say that the "required = true"-attributes can generally be removed here and elsewhere in the endpoints of this branch. 


https://docs.spring.io/spring-framework/docs/3.0.x/javadoc-api/org/springframework/web/bind/annotation/RequestParam.html#:~:text=org.springframework.web.bind.annotation%20Annotation%20Type%20RequestParam%20%40Target%28value%3DPARAMETER%29%20%40Retention%28value%3DRUNTIME%29%20%40Documentedpublic%20%40interface,annotated%20handler%20methods%20in%20Servlet%20and%20Portlet%20environments.